github: add issue in form format

Deflaimun opened the pull request:
Resolves #319 

To see how it looks, use one of my repositories as an example
https://github.com/Deflaimun/rest-table/issues/new/choose

micheleRP commented:
@Deflaimun, @Feediver1: I'll do a copyedit on this text, but I want to make sure I understand the different issue types first. Currently, we have 2 doc issue types: 
- Documentation request
- Report a security vulnerability 

With these new forms, we also have 2 types: 
- **Documentation request** looks similar to the current type
- **New feature** says it's to suggest a new feature but is strictly used internally by Redpanda

If **New feature** is internal only (guessing for confidential information/conversation), can we rename it **Internal Redpanda doc request**, and can those internal issues be hidden?

Also, are we removing the issue for reporting security vulnerabilities? Have we ever had anyone use that?

Deflaimun commented:
@micheleRP the issue template for "reporting security vulnerability " is something that we get by default from git.
It's meant to capture vulnerabilities within the repository. This happens because of some mandatory settings that our repo must follows. It should be used to report something wrong with the framework for example. Git uses this template to create automatic security updates for us.

So in reality we only had 1 template for everything. That template had a lot of "if new feature do this". So instead I split it into two different templates where we can enforce required fields differently. 

> If New feature is internal only (guessing for confidential information/conversation), can we rename it Internal Redpanda doc request, and can those internal issues be hidden?

My initial idea was to make one template for general changes to docs and one specific for new features. Since customers won't submit new features, I added a text to make sure the correct stakeholders use them. As for the naming, whatever works for everyone. And AFAIK we can't hide those issues with a public repository.

micheleRP commented:
Ah, ok, I see. Is it correct that only Redpanda employees can create new feature issues? If so, here's what I recommend (and I'll add copy edits for each type as PR suggestions): 

📚 Documentation update
Suggest a change to existing Redpanda documentation.
🚀 New feature
Request documentation for a new feature. (Redpanda employees only.)

micheleRP commented:
If we want to keep all doc requests on one form, here's a suggestion for the top-level selection:

📚 Documentation
Create issues for doc bugs, improvements, or content gaps. Redpanda employees can also create issues for documenting new features. 

Here's a suggestion for inside the form. _This would just have 2 boxes: 1 for updating doc and 1 for new features. I like having the description for each issue type inside the Write box, like how Paulo shows it in [New feature additional info](https://github.com/Deflaimun/rest-table/issues/new?assignees=&labels=New+feature&projects=&template=new_feature.yml), but I don't like how it disappears as soon as you start to write, so I guess the description for each would go above the box._ 

Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and you can ask for guidance on our [community Slack docs channel](https://redpandacommunity.slack.com/archives/C048589SBEZ).

Create issues for doc bugs, improvements, or content gaps. Redpanda employees can also create issues for documenting new features. 

**Update documentation** 
- Describe how the current doc is incorrect, incomplete, or unclear, and describe how to fix it. 
- Include as much information as possible, including the URL of the doc needing update and any links to slack conversations or more details about the issue. 

**For new features**
- What problem does this feature solve? 
- How does it work (concepts and procedures)?
- What are the use cases and benefits?
- What release will it be available?
- Who are the SMEs for approving the feature doc? 
- Include any new or changing config flags, rpk commands, or metrics. 
- Include links to RFCs, PRDs, blogs, KIPs, related issues, or Slack conversations.

JakeSCahill commented:
The `redpanda` repo also has a link to Slack. Maybe, we could use that too.

![2023-06-23_12-29-40](https://github.com/redpanda-data/documentation/assets/45230295/93783c2e-5076-4e6a-8d55-889273b76011)


`ISSUE-TEMPLATE/config.yaml`
```yaml
blank_issues_enabled: true
contact_links:
  - name: 👋 Slack
    url: https://redpanda.com/slack
    about: Chat about all things Redpanda. Join the conversation!
```

JakeSCahill commented:
Seeing all the options in one form makes me think that two separate templates may work best.

If a user wants to report a simple mistake in the docs, having lots of text about new features adds unnecessary cognitive load. If we have two templates, users make a choice at the start of the journey and the form can cater to their needs in a more concise way.

We might also want to consider a template for website update requests (style changes, modal changes, etc...).

Deflaimun commented:
```
Update documentation

Describe how the current doc is incorrect, incomplete, or unclear, and describe how to fix it. Include as much information as possible.
Include the URL of the doc needing update and any links to more details or Slack conversations about the issue.
New feature

What problem does this feature solve?
How does it work (concepts and procedures)?
What are the use cases and benefits?
What release will it be available?
Who are the SMEs for approving the feature doc?
Include any new or changing config flags, rpk commands, or metrics.
Include links to RFCs, PRDs, blogs, KIPs, related issues, or Slack conversations.
```

@micheleRP which of those do you think should be mandatory?

micheleRP commented:
> ```
> Update documentation
> 
> Describe how the current doc is incorrect, incomplete, or unclear, and describe how to fix it. Include as much information as possible.
> Include the URL of the doc needing update and any links to more details or Slack conversations about the issue.
> New feature
> 
> What problem does this feature solve?
> How does it work (concepts and procedures)?
> What are the use cases and benefits?
> What release will it be available?
> Who are the SMEs for approving the feature doc?
> Include any new or changing config flags, rpk commands, or metrics.
> Include links to RFCs, PRDs, blogs, KIPs, related issues, or Slack conversations.
> ```
> 
> @micheleRP which of those do you think should be mandatory?

I think they should all be mandatory. Then we'll get better info for our doc issues! Users can write N/A if it doesn't apply; e.g., if there are no new commands/metrics. I guess then we should rewrite the list as: 
- What problem does this feature solve?
- Describe use cases and benefits.
- List all relevant concepts and procedures to explain how it works.
- In what release will it be available?
- Who are the SMEs for approving the feature doc?
- Does this include any new or changing config flags, rpk commands, or metrics?
- Include links to more info: RFCs, PRDs, blogs, KIPs, related issues, or Slack conversations.

Feediver1 commented:
+1 on Michele's descriptions. Only suggestion there is to make the URL mandatory for documentation update suggestions.

Feediver1 commented on .github/ISSUE_TEMPLATE/docs_template.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: markdown
+    attributes:
+      value: |
+        This template is strictly used by customer requests. If you need to fill a new feature, consider using New Feature.
```
```suggestion
        **This template is strictly for customer-requested documentation updates. If you want to request documentation for a new feature, use the New Feature option.**
```

Feediver1 commented:
From Michele's suggestion:

Update documentation

Provide specific details about how the current documentation is incorrect, incomplete, or unclear, and also describe how to fix it. Include as much information as possible.
(Required) Include the URL of the documentation page that requires an update and any links to additional details or Slack conversations about the issue.

New feature

What problem does this feature solve?
Are there any prerequisites or limitations that users need to be aware of?
How does it work (concepts and procedures)?
What are the use cases and benefits?
In which release will it be available?
Who are the SMEs for approving the feature doc?
Include any new or changing config flags, rpk commands, or metrics.
Include links to RFCs, PRDs, blogs, KIPs, related issues, or Slack conversations.

Feediver1 commented on .github/ISSUE_TEMPLATE/02_new_feature.yml:
```diff
@@ -0,0 +1,60 @@
+
+        Each Issue will drive the new content in GitHub for the respective platform or cloud doc. 
+
+        This template is strictly used internally by Redpanda. If a documentation needs additional clarification or you're reporting a bug, consider using Documentation Request.
```
```suggestion
        This template is strictly used internally by Redpanda. If documentation needs additional clarification or you're reporting a bug, file a Documentation Request issue.
```

Feediver1 commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+name: 📚 Documentation request
```
I think this should be "Documentation update request"

Deflaimun replied:
```suggestion
name: 📚 Documentation update request
```

Feediver1 commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: markdown
+    attributes:
+      value: |
+        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and you can ask for guidance on our [community Slack docs channel](https://redpandacommunity.slack.com/archives/C048589SBEZ).
```
```suggestion
        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and ask for guidance on our [community Slack docs channel](https://redpandacommunity.slack.com/archives/C048589SBEZ).
```

Feediver1 commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+      value: |
+        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and you can ask for guidance on our [community Slack docs channel](https://redpandacommunity.slack.com/archives/C048589SBEZ).
+
+        Create issues for doc bugs, improvements, or content gaps. If you need to fill a new feature, consider using the New Feature template.
```
```suggestion
        Create issues for doc bugs, improvements, or content gaps. If you need to request a new feature, consider using the New Feature template.
```

Feediver1 commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+      description: If applicable, provide the link to the Redpanda Community Slack conversation.
+  - type: markdown
+    attributes:
+      value: Need to talk about anything else? You can reach us directly at [Redpanda Community Slack](https://rpnda.co/slack).
```
```suggestion
      value: Need to ask questions about Redpanda documentation? You can reach us directly at [Redpanda Community Slack](https://rpnda.co/slack).
```

Deflaimun commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: markdown
+    attributes:
+      value: |
+        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and ask for guidance on our [community Slack docs channel](https://redpandacommunity.slack.com/archives/C048589SBEZ).
```
```suggestion
        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and ask for guidance on our [community Slack docs channel](https://redpanda.com/slack).
```

Deflaimun commented:
Hello reviewers! We're changing our issues templates and would like to know your feedback on it? Do you like it? Want to change something? 

Ultimately you'll be the users for this product so it's important that we hear your feedback.

Thanks!

mattschumpert commented on .github/ISSUE_TEMPLATE/02_new_feature.yml:
```diff
@@ -0,0 +1,60 @@
+  - type: textarea
+    attributes:
+      label: Code changes
+      description: Does this include any new or changing config flags, rpk commands, or metrics?
```
This one is confusing. All new features have 'code changes'.   Do you mean 'incompatible changes'?  

Even new features with config flags are not incompatible changes, only removal of configs/metrics/commands or APIs.

micheleRP replied:
@mattschumpert: What if we remove this **Code changes** item and instead tack its description onto the **How it works** item above? e.g.,
**How it works**
How does it work (concepts and procedures)? Include any new or changing config flags, rpk commands, or metrics.

micheleRP replied:
Actually, for a description, how about: 

List new concepts and the procedures for using the feature. Include any new or changing config flags, rpk commands, or metrics.

Deflaimun replied:
I like this one @micheleRP Let me implement that. 
It should read now: 

> **How it works**
> List new concepts and the procedures for using the feature. Include any new or changed config flags, rpk commands, or metrics.

mattschumpert reviewed (approved):
Generally seems like an improvement to have these structures.  Left one comment.

Feediver1 reviewed (approved):
go with any final edits from @micheleRP 
Otherwise, LGTM

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: markdown
+    attributes:
+      value: |
+        Help us improve Redpanda documentation! You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and ask for guidance on our [community Slack docs channel](https://redpanda.com/slack).
+
+        Create issues for doc bugs, improvements, or content gaps. If you need to request a new feature, consider using the New Feature template.
```
```suggestion
        Help us improve Redpanda documentation! Create issues for doc bugs, improvements, or content gaps. You can search [open doc issues](https://github.com/redpanda-data/documentation/issues), and ask for guidance on our [community Slack docs channel](https://redpanda.com/slack). To request documentation for a new feature, use the New Feature template.
```

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+        Create issues for doc bugs, improvements, or content gaps. If you need to request a new feature, consider using the New Feature template.
+  - type: textarea
+    attributes:
+      label: Issue Description
```
```suggestion
      label: Issue description
```

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: textarea
+    attributes:
+      label: Updates to existing documentation
+      description: Include the URL of the documentation page that requires an update and any links to additional details or Slack conversations about the issue.
```
```suggestion
      description: Provide the URL of the page that requires an update and any links to additional information.
```

micheleRP replied:
```suggestion
      description: Provide the URL of the page that requires an update.
```

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+  - type: textarea
+    attributes:
+      label: Link to Redpanda Community Slack conversation
+      description: If applicable, provide the link to the Redpanda Community Slack conversation.
+  - type: markdown
```
```suggestion
      description: If applicable, provide links to any Redpanda Slack conversations about the issue.
  - type: markdown
```

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+      description: If applicable, provide the link to the Redpanda Community Slack conversation.
+  - type: markdown
+    attributes:
+      value: Need to ask questions about Redpanda documentation? You can reach us directly at [Redpanda Community Slack](https://rpnda.co/slack).
```
```suggestion

```
suggest removing this altogether. We say the same thing at the top of the issue.

Deflaimun replied:
Removing..

micheleRP commented on .github/ISSUE_TEMPLATE/01_doc_request.yml:
```diff
@@ -0,0 +1,29 @@
+      required: true  
+  - type: textarea
+    attributes:
+      label: Link to Redpanda Community Slack conversation
```
```suggestion
      label: Link to Redpanda Slack conversation
```
suggest removing Community, because we don't want internal Redpanda folks to think this form is only for issues coming from external users

micheleRP commented on .github/ISSUE_TEMPLATE/02_new_feature.yml:
```diff
@@ -0,0 +1,56 @@
+  - type: markdown
+    attributes:
+      value: |
+        Create a new issue for new features to be documented. Provide as much information as possible. 
+
+        Documentation Issues must be distinct and not overlap. 
+
+        If filing an Issue for a feature that spans platform and cloud, you must file two separate Issues. 
+
+        Each Issue will drive the new content in GitHub for the respective platform or cloud doc. 
+
+        This template is strictly used internally by Redpanda. If documentation needs additional clarification or you're reporting a bug, file a Documentation Request issue.
```
```suggestion
        Create an issue for new features to be documented. Provide as much information as possible. This template is strictly used internally by Redpanda. 
```

micheleRP commented on .github/ISSUE_TEMPLATE/02_new_feature.yml:
```diff
@@ -0,0 +1,56 @@
+        This template is strictly used internally by Redpanda. If documentation needs additional clarification or you're reporting a bug, file a Documentation Request issue.
+  - type: textarea
+    attributes:
+      label: Feature Description
```
```suggestion
      label: Feature description
```

micheleRP commented on .github/ISSUE_TEMPLATE/02_new_feature.yml:
```diff
@@ -0,0 +1,56 @@
+      required: true
+  - type: textarea
+    attributes:
+      label: Prerequisites 
```
Can you please move Prerequisites down under Use cases?

Feediver1 commented on modules/get-started/pages/licensing/overview.adoc:
```diff
@@ -109,6 +109,10 @@ The following table lists the enterprise features for Redpanda and how Redpanda
 
+| xref:develop/manage-topics/cloud-topics.adoc[Cloud Topics]
+| A Redpanda topic type that uses durable object storage as the primary backing store instead of local disk replication.
+| Cloud Topics cannot be created. Existing Cloud Topics cannot be modified. Additional partitions cannot be added to existing Cloud Topics. Major upgrades are blocked when in a violation state.
```
```suggestion
| New Cloud Topics cannot be created. Existing Cloud Topics cannot be modified (including the addition or modification of partitions for existing Cloud Topics). Major upgrades are blocked when in a violation state.
```

Feediver1 commented on modules/get-started/pages/licensing/overview.adoc:
```diff
@@ -109,6 +109,10 @@ The following table lists the enterprise features for Redpanda and how Redpanda
 | Records detailed logs of cluster activities for compliance and monitoring.
 | Read access to the audit log topic is denied, but logging continues.
 
+| xref:develop/manage-topics/cloud-topics.adoc[Cloud Topics]
```
```suggestion
| xref:develop:manage-topics/cloud-topics.adoc[Cloud Topics]
```

Feediver1 commented on modules/get-started/pages/licensing/overview.adoc:
```diff
@@ -109,6 +109,10 @@ The following table lists the enterprise features for Redpanda and how Redpanda
 
+| xref:develop/manage-topics/cloud-topics.adoc[Cloud Topics]
+| A Redpanda topic type that uses durable object storage as the primary backing store instead of local disk replication.
+| New Cloud Topics cannot be created. Existing Cloud Topics cannot be modified (including the addition or modification of partitions for existing Cloud Topics). Major upgrades are blocked when in a violation state.
```
```suggestion
| New Cloud Topics cannot be created. Existing Cloud Topics cannot be modified, including adding or modifying partitions. Major upgrades are blocked when in a violation state.
```

micheleRP reviewed (approved):
lgtm
